pages/common/histexpand.md

```diff
@@ -0,0 +1,36 @@
+# history expansion
+
+> Reuse and expand the shell history in bash or zsh.
+> More information: <https://www.gnu.org/software/bash/manual/html_node/History-Interaction.html>.
+
+- Run the previous command:
+
+`!!`
+
+- Run the previous command as root:
+
+`sudo !!`
+
+- Run a command with the last argument of the previous command:
+
+`{{command}} !$`
+
+- Run a command with the first argument of the previous command:
+
+`{{command}} !^`
+
+- Run a command `n` lines back in the history:
+
+`!-{{n}}`
+
+- Run the most recent command starting with `string`:
+
+`!{{string}}`
+
+- Run the previous command, replacing `string1` with `string2`:
+
+`^{{string1}}^{{string2}}^`
+
+- Print a command `n` lines back without running it:
```

Comment:
Same as above. Also, maybe move it to be next to the `!-{{n}}` example?
```suggestion
- Print the command `n` lines back without running it:
```

Reply:
I'm not sure. `:p` applies to all expansions, I just took `!-n` as an example, so I'd say they aren't really related. That's also the reason I put them at the end of the page.

Reply:
I  that case I'd suggest trying a different description that makes the generality more evident. The way it is written, it seems to be directly related to the example above.

Reply:
What would you suggest?

Reply:
https://github.com/tldr-pages/tldr/pull/5455#discussion_r606834489 ?

Reply:
Yep, that is a bit more verbose, but conveys the right meaning/context IMO.